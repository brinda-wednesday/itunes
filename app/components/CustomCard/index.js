/**
 *
 * CustomCard
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import T from '@components/T';
import { Card } from 'antd';
import styled from 'styled-components';

const StyledCard = styled(Card)`
  && {
    margin: 20px 0;
    max-width: ${(props) => props.maxWidth ?? 900}px;
    color: grey;
  }
`;

export function CustomCard({ name, stargazersCount, fullName }) {
  return (
    <StyledCard data-testid="custom-card">
      <T id="repository_name" data-testid="repo-name" values={{ name: name }} />
      <T id="repository_full_name" data-testid="full-name" values={{ fullName: fullName }} />
      <T id="repository_stars" data-testid="stargazer-count" values={{ stars: stargazersCount }} />
    </StyledCard>
  );
}

CustomCard.propTypes = {
  name: PropTypes.string,
  stargazersCount: PropTypes.number,
  fullName: PropTypes.string
};

export default CustomCard;
