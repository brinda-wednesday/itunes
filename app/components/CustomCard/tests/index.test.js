/**
 *
 * Tests for CustomCard
 *
 */

import React from 'react';
// import { fireEvent } from '@testing-library/dom'
import { renderWithIntl, renderProvider } from '@utils/testUtils';
import CustomCard from '../index';

describe('<CustomCard />', () => {
  it('should render and match the snapshot', () => {
    const { baseElement } = renderWithIntl(<CustomCard />);
    expect(baseElement).toMatchSnapshot();
  });

  it('should contain 1 CustomCard component', () => {
    const { getAllByTestId } = renderWithIntl(<CustomCard />);
    expect(getAllByTestId('custom-card').length).toBe(1);
  });

  it('should render repo name', () => {
    const name = 'xyz';
    const { getAllByTestId, getByTestId } = renderProvider(<CustomCard name={name} />);
    expect(getAllByTestId('repo-name').length).toBe(1);
    expect(getByTestId('repo-name').textContent).toBe('Repository Name: xyz');
  });

  it('should render full name', () => {
    const fullName = 'abc';
    const { getAllByTestId, getByTestId } = renderProvider(<CustomCard fullName={fullName} />);
    expect(getAllByTestId('full-name').length).toBe(1);
    expect(getByTestId('full-name').textContent).toBe('Repository full name: abc');
  });

  it('should render stargazers count', () => {
    const stargazersCount = 1234;
    const { getAllByTestId, getByTestId } = renderProvider(<CustomCard stargazersCount={stargazersCount} />);
    expect(getAllByTestId('stargazer-count').length).toBe(1);
    expect(getByTestId('stargazer-count').textContent).toBe('Repository stars: 1234');
  });
});
