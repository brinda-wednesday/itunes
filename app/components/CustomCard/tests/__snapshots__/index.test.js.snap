// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CustomCard /> should render and match the snapshot 1`] = `
<body>
  <div>
    <div
      class="ant-card ant-card-bordered CustomCard__StyledCard-sc-19zlds6-0 cgGzzh"
      data-testid="custom-card"
    >
      <div
        class="ant-card-body"
      >
        <p
          class="T__StyledText-gjlic1-0 kYcyRq"
          data-testid="repo-name"
        >
          Repository Name: 
        </p>
        <p
          class="T__StyledText-gjlic1-0 kYcyRq"
          data-testid="full-name"
        >
          Repository full name: 
        </p>
        <p
          class="T__StyledText-gjlic1-0 kYcyRq"
          data-testid="stargazer-count"
        >
          Repository stars: 
        </p>
      </div>
    </div>
  </div>
</body>
`;
